packages/flutter/test/widgets/scrollable_selection_test.dart

```diff
@@ -0,0 +1,641 @@
+
+  testWidgets('select to scroll by dragging selection handles forward', (WidgetTester tester) async {
+    final ScrollController controller = ScrollController();
+    await tester.pumpWidget(MaterialApp(
+      home: SelectionArea(
+        selectionControls: materialTextSelectionControls,
+        child: ListView.builder(
+          controller: controller,
+          itemCount: 100,
+          itemBuilder: (BuildContext context, int index) {
+            return Text('Item $index');
+          },
+        ),
+      ),
+    ));
+    await tester.pumpAndSettle();
+
+    // Long press to bring up the selection handles.
+    final RenderParagraph paragraph0 = tester.renderObject<RenderParagraph>(find.descendant(of: find.text('Item 0'), matching: find.byType(RichText)));
+    final TestGesture gesture = await tester.startGesture(textOffsetToPosition(paragraph0, 2));
+    addTearDown(gesture.removePointer);
+    await tester.pump(const Duration(milliseconds: 500));
+    await gesture.up();
+    expect(paragraph0.selections[0], const TextSelection(baseOffset: 0, extentOffset: 4));
+
+    final List<TextBox> boxes = paragraph0.getBoxesForSelection(paragraph0.selections[0]);
+    expect(boxes.length, 1);
+    final Offset handlePos = globalize(boxes[0].toRect().bottomRight, paragraph0);
+    await gesture.down(handlePos);
+
+    expect(controller.offset, 0.0);
+    double previousOffset = controller.offset;
+
+    await gesture.moveTo(tester.getBottomRight(find.byType(ListView)));
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 1));
+    expect(controller.offset > previousOffset, isTrue);
+    previousOffset = controller.offset;
+
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 1));
+    expect(controller.offset > previousOffset, isTrue);
+
+    // Scroll to the end.
+    await tester.pumpAndSettle(const Duration(seconds: 1));
+    expect(controller.offset, 4200.0);
+    final RenderParagraph paragraph99 = tester.renderObject<RenderParagraph>(find.descendant(of: find.text('Item 99'), matching: find.byType(RichText)));
+    final RenderParagraph paragraph98 = tester.renderObject<RenderParagraph>(find.descendant(of: find.text('Item 98'), matching: find.byType(RichText)));
+    final RenderParagraph paragraph97 = tester.renderObject<RenderParagraph>(find.descendant(of: find.text('Item 97'), matching: find.byType(RichText)));
+    final RenderParagraph paragraph96 = tester.renderObject<RenderParagraph>(find.descendant(of: find.text('Item 96'), matching: find.byType(RichText)));
+    expect(paragraph99.selections[0], const TextSelection(baseOffset: 0, extentOffset: 7));
+    expect(paragraph98.selections[0], const TextSelection(baseOffset: 0, extentOffset: 7));
+    expect(paragraph97.selections[0], const TextSelection(baseOffset: 0, extentOffset: 7));
+    expect(paragraph96.selections[0], const TextSelection(baseOffset: 0, extentOffset: 7));
+    await gesture.up();
+  });
```

Comment:
Do the selection handles show up in reaction to touch events even on desktop platforms?

I think this might be how EditableText works now too, no need to change it if so.

Reply:
yes if you creates EditableText with materialTextSelectionControls, it will show up for desktop too